python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/search.py

```diff
@@ -176,19 +186,53 @@ def list_data_product_assets(
         except GraphError as e:
             raise Exception("Unable to execute listDataProductAssets query") from e
         page_results = []
-        for result in response["listDataProductAssets"]["searchResults"]:
+        page_results = self._get_data_collection_page_results(
+            response, "listDataProductAssets"
+        )
+
+        return SearchResponse(
+            total_results=response["listDataProductAssets"]["total"],
+            page_results=page_results,
+        )
+
+    def list_database_tables(
+        self, urn: str, count: int, start: int = 0
+    ) -> SearchResponse:
+        variables = {
+            "urn": urn,
+            "start": start,
+            "count": count,
+        }
+
+        try:
+            response = self.graph.execute_graphql(
+                self.get_database_tables_query, variables
+            )
+        except GraphError as e:
+            raise Exception("Unable to execute listDatabaseEntities query") from e
+        page_results = self._get_data_collection_page_results(
```

Comment:
whitespace suggestion

```suggestion

        page_results = self._get_data_collection_page_results(
```